### ISSUE TITLE
IE8 compatibility: RGBA color plus separate color property creates invalid result

### ISSUE THREAD

**Issue by stevendesu** [jakubpawlowicz/clean-css#711]
## What's your input CSS and expected output? ##

**Input CSS:**
```css
.element {
    border: solid rgba(0,0,0,0);
    border-color: transparent;
}
```

**Expected Output:**
```css
.element{border:solid transparent}
```

**Note:** In all common browsers whichever rule is listed second will take priority. Had `border-color` been listed before `border` I would expect to see `rgba(0,0,0,0)`

**Actual Output:**
```css
.element{border:solid rgba(0,0,0,0) transparent}
```

**Note:** This is invalid CSS

## Do you use clean-css directly or through any of the plugins? ##

I'm running clean-css through the [gulp-minify-css plugin](https://github.com/murphydanger/gulp-minify-css)

## What options do you pass to clean-css / the plugin? ##

```json
{
    "compatibility": "ie8"
}
```

**Note:** I discovered that *without* this option the issue does not occur, however this option is necessary for my use-case.

## What version of clean-css do you use? ##

3.4.8

## What operating system do you use? ##

root@C9-workspace:~/cms# uname -a
Linux C9-workspace 3.13.0-37-generic #64-Ubuntu SMP Mon Sep 22 21:28:38 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

**Comment by jakubpawlowicz** [comment]
Thanks @stevendesu for outstanding bug report! This is incorrect indeed. Please correct me if I'm wrong but IE8 compatible output should be the same as input, since IE8 does not support `rgba`.

**Comment by stevendesu** [comment]
To be honest, I'm not 100% sure what the expected output would be for IE8 compatible mode :smile: I suppose it depends on the intended function.

The primary reason I'm sticking to IE8 support is that this customer has 2% of their clients using IE8 which they said was too large to completely disregard, but they were willing to accept a few quirks here or there (rounded corners not being round, etc) - so I've been for the most part ignoring IE8 compatibility then testing in IE8 to make sure the site at least isn't broken. When I ran my code through clean-css without IE8 compatibility it broke some rules that were otherwise working.

If the goal is "ignore any rules that you can't support / don't understand", you should probably leave the whole thing uncompressed:

```css
.element{border:solid rgba(0,0,0,0);border-color:transparent}
```

If the goal is "remove any rules that you can't support", you should rip out the rbga but leave transparent intact:

```css
.element{border:solid transparent}
```

If the goal is "convert all rules to be ie8 compatible", you should convert rgba to rgb with a special case to handle 0 for the alpha (converting it to "transparent", which is a valid CSS2 color)

```css
.element{border:solid transparent} /* would have been #000 if not for 0 alpha */
```

For the sake of maintaining the same functionality between compressed and uncompressed versions, I'm tempted to go with number one. Let IE8 choke on the first rule but at least read the second one correctly.

**Comment by jakubpawlowicz** [comment]
Yes, that's how it should work - i.e. support IE8 and greater just as a **developer** intended. clean-css has no intent to backport any styling.

**Comment by jakubpawlowicz** [comment]
It's fixed in 3.4.9.
